platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,210 @@
+import org.osgi.framework.ServiceEvent;
+import org.osgi.framework.ServiceReference;
+import org.osgi.framework.hooks.service.EventListenerHook;
+import org.osgi.framework.hooks.service.ListenerHook.ListenerInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ErrorPageInjector implements EventListenerHook {
+
+  private final String errorPagePath = "/ErrorServlet";
+  private final Map<String, String> errorCodesMap;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ErrorPageInjector.class);
+
+  public ErrorPageInjector() {
+    errorCodesMap =
+        Stream.of("400", "401", "403", "404", "405", "406", "500", "501", "502", "503", "504")
+            .collect(
+                Collectors.collectingAndThen(
+                    Collectors.toMap(Function.identity(), x -> errorPagePath),
+                    Collections::unmodifiableMap));
+  }
+
+  Optional<BundleContext> getContext() {
+    final Bundle cxfBundle = FrameworkUtil.getBundle(ErrorPageInjector.class);
+    return (cxfBundle != null) ? Optional.of(cxfBundle.getBundleContext()) : Optional.empty();
+  }
+
+  @Override
+  public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo>> listeners) {
+    if (event.getType() == ServiceEvent.REGISTERED) {
+      Bundle refBundle = event.getServiceReference().getBundle();
+      BundleContext bundlectx = refBundle.getBundleContext();
+      Object service = bundlectx.getService(event.getServiceReference());
+      if (service instanceof ServletContext) {
+        injectErrorPage((ServletContext) service, refBundle);
+      }
+    }
+  }
+
+  public void init() {
+    checkForMissedServletContexts();
+  }
+
+  private void checkForMissedServletContexts() {
+
+    Optional<BundleContext> optionalBundleContext = getContext();
+    if (!optionalBundleContext.isPresent()) {
+      LOGGER.warn(
+          "Problem retrieving the bundle `Platform :: Error :: Page Injector` after it's just been initialized");
+      return; // This should never happen
+    }
+
+    //  Get all registered servletContexts in OSGI
+    final BundleContext context = optionalBundleContext.get();
+    getServiceReferences(context)
+        .ifPresent(e -> Arrays.asList(e).forEach(this::addMissedErrorPage));
+  }
+
+  @SuppressWarnings("unchecked" /* cast of Object[] to ServiceRefrence */)
+  private Optional<ServiceReference<ServletContext>[]> getServiceReferences(BundleContext context) {
+    try {
+      return Optional.ofNullable(
+          (ServiceReference<ServletContext>[])
+              context.getAllServiceReferences(
+                  null, "(" + Constants.OBJECTCLASS + "=javax.servlet.ServletContext)"));
+
+    } catch (InvalidSyntaxException | ClassCastException e) {
+      LOGGER.error(
+          "Problem getting ServletContexts from OSGI, One of the servlets running might print stack traces to the browser. A system restart is recommended. See debug logs for additional details.");
```

Comment:
✏️ Seems like after OSGI it should be a period instead of a comma, but a nit and may be just how I'm reading it.

Reply:
Nice catch,  either way something needs to change since One is capitalized